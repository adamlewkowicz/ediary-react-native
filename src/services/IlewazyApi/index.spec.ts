import { IlewazyApi } from '.';
import ilewazyResponseMock from './__mocks__/ilewazy.json';

describe('IlewazyApi', () => {

  let ilewazyApi: IlewazyApi;

  beforeEach(() => ilewazyApi = new IlewazyApi());

  describe('findByName()', () => {

    it('should normalize products correctly', async () => {
      const name = 'apple';
      const fetchMock = fetch as jest.Mock;
    
      fetchMock.mockImplementationOnce(async () => ({
        ok: true,
        json: async () => ilewazyResponseMock
      }));
    
      const [
        firstNormalizedProduct,
        secondNormalizedProduct,
      ] = await ilewazyApi.findByName(name);
    
      expect(fetchMock).toHaveBeenCalledTimes(1);
      expect(firstNormalizedProduct).toMatchSnapshot();
      expect(secondNormalizedProduct).toMatchSnapshot();
    });

  });

});