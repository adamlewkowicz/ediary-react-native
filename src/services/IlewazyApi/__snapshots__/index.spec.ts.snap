// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IlewazyApi findByName() should normalize products correctly 1`] = `
Object {
  "_id": "16390",
  "images": Array [
    "http://static.ilewazy.pl/dziennik/470/bagatelka-ciastka-jablko-z-cynamonem-1.jpg",
    "http://static.ilewazy.pl/dziennik/470/bagatelka-ciastka-jablko-z-cynamonem-2.jpg",
  ],
  "macro": Object {
    "carbs": 104,
    "fats": 7.8,
    "kcal": 365,
    "prots": 7,
  },
  "name": "Bagatelka owocowa, kruche ciasteczka z nadzieniem jabłkowym z cynamonem",
  "portion": 16,
  "portions": Array [
    Object {
      "type": "piece",
      "unit": "g",
      "value": 16,
    },
    Object {
      "type": "portion",
      "unit": "g",
      "value": 48,
    },
  ],
}
`;

exports[`IlewazyApi findByName() should normalize products correctly 2`] = `
Object {
  "_id": "17551",
  "images": Array [
    "http://static.ilewazy.pl/dziennik/470/baton-marchew-jablko-pomarancza-freeyu-1.jpg",
  ],
  "macro": Object {
    "carbs": 135,
    "fats": 3.6,
    "kcal": 333,
    "prots": 1.4,
  },
  "name": "Baton jabłkowo-pomarańczowy z marchwią",
  "portion": 40,
  "portions": Array [
    Object {
      "type": "piece",
      "unit": "g",
      "value": 40,
    },
  ],
}
`;
